TechnicalDocuments/FastlaneMatch.md

```diff
@@ -0,0 +1,147 @@
+
+Since the files in the GitHub repository are encrypted they cannot be updated directly. Fastlane Match provides a couple of helper functions. To clone the repository, open a ruby console and call `Match::GitHelper.clone`. Babylon specific credentials can be found in the iOS 1Password vault.
+
+```shell
+$ bundle console
+irb(main):001:0> require 'match'
+=> true
+irb(main):002:0> branch = 'build_target'
+=> "build_target"
+irb(main):003:0> git_url = 'https://github.com/path/to/fastlane/match/repo'
+=> "https://github.com/path/to/fastlane/match/repo"
+irb(main):004:0> shallow_clone = false
+=> false
+irb(main):005:0> password = 'long-and-super-secret'
+=> "long-and-super-secret"
+irb(main):006:0> workspace = Match::GitHelper.clone(git_url, shallow_clone, manual_password: password, branch: branch)
+[14:22:10]: Cloning remote git repo…
+[14:22:10]: If cloning the repo takes too long, you can use the `clone_branch_directly` option in match.
+[14:22:12]: Checking out branch build_target…
+[14:22:12]: 🔓  Successfully decrypted certificates repo
+=> "/var/folders/8k/r0x0ys_927q8vq5_tq01ntjd2b03m3/T/d20190228-92193-16w4fz2"
+```
+
+Open another terminal session and navigate to the folder name returned by `Match::GitHelper.clone`. Once all edits are complete, return to the ruby console.
+
+```shell
+irb(main):007:0> commit_message = 'Updated provisioning profiles to be great again.'
+=> "Updated provisioning profiles to be great again.”
+irb(main):008:> Match::GitHelper.commit_changes(workspace, commit_message, git_url, branch)
+[14:28:44]: 🔒  Successfully encrypted certificates repo
+[14:28:44]: Pushing changes to remote git repo…
+=> nil
+```
+
+It can be convenient to store the password in the environment hash
+
+```shell
+irb(main):008:> ENV[“MATCH_PASSWORD"] = 'long-and-super-secret'
+```
+
+The password itself can be found in the iOS team 1Password vault.
+
+Certificates are stored in three folders
+
+```shell
+ certs/development
+ certs/distribution
+ certs/enterprise
+```
+
+It might be necessary to manually make a certificate available in both the enterprise and distribution folder. Copying the files did not work (February 2019), but adding a soft link did.
+
+Fastlane Math expects the filename for the .p12 and .cer to be `certificate id`.p12 and `certificate id`.cer, whereas manually generated certificates that have been downloaded from the developer portal are named `team id`.cer and `team id`.p12.
+
+Fastlane has a utility for interacting with Apple Developer Portal and the App Store Connect API called [spaceship](https://github.com/fastlane/fastlane/tree/master/spaceship). There were some problems (March 2019) with running `bundle exec fastlane spaceship`. It might be necessary to manually install some gems in the `~/.fastlane` directory.
```

Comment:
Please consider adding another title to make this section stand out more, e.g.

```
## Accessing Apple Developer Portal with fastlane spaceship
```

Reply:
I checked now and for me only missing `gem` was `pry`. Is that the issue you encountered?

If yes we could add `pry` to our `Gemfile` and remove this section:

> There were some problems (March 2019) with running `bundle exec fastlane spaceship`. It might be necessary to manually install some gems in the `~/.fastlane` directory.

My error logs after running `bundle exec fastlane spaceship`:
```console
Could not find gem 'pry'

If you installed spaceship using `sudo gem install spaceship` run
  sudo gem install pry
to install the missing gem

If you use a Gemfile add this to your Gemfile:
  gem 'pry'
and run `bundle install`
```

Reply:
Unfortunately, adding `pry` to my Gemfile gave version errors. Have you tested if it works for you @mr-v ?

Reply:
Yes, I can login.

Reply:
I can login after installing `pry` manually, but adding `pry` to the Gemfile did not work.

Reply:
@zzcgumn Are you using zsh, especially oh-my-zsh with the bundler plugin? If so, it automatically wraps calls to some gems into `bundle exec`, and `pry` is one of them by default. Meaning that running `pry` in such a zsh shell will actually run `bundle exec pry`, then seing that you don't have `pry` in your `Gemfile`, will abort... maybe that's related?

(FYI In such cases you can use `unbundled_pry` to run the raw `pry` command without zsh wrapping it in `bundle exec` for you, or you can edit your `.zshrc` to add `UNBUNDLED_COMMANDS=(pry)` before the `plugins=...` line)

Reply:
>  I'd keep instructions simple and only add problems note and workaround information when it becomes a more widespread issue.

I agree with @mr-v on that. As these instructions are far from what we need to deal with on a regular basis, I'd either move it to another document about troubleshooting signing issues or would hide it under `summary` as I suggested before.